### PR TITLE
Satellite support

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: pkg/k8s_utils/k8s_client.go [hunk 1]
```diff
@@ -30,12 +39,64 @@ type KubernetesClient struct {
...
+
+	kc.clientset = clientset
+	kc.logger = logger
+	kc.namespace = namespace
+	return kc, nil
+}
+
+// getNameSpace ...
+func getNameSpace(logger *zap.Logger) (string, error) {
+	// Reading the namespace in which the pod is deployed
```
Comment (skorati) [inline pkg/k8s_utils/k8s_client.go:87]: `Reading the namespace in which the pod is deployed`
which pod?

**Review by skorati — APPROVED** [review]
looks good.

File: pkg/k8s_utils/k8s_client.go (hunk 1 above)
Comment (GunaKKIBM) [reply pkg/k8s_utils/k8s_client.go:87]: CSI driver

**Review by arahamad — APPROVED** [review]
lgtm, please verify changes on iks, roks, ipi and satellite(although forming private endpoint will not work, we need to add more logic to fetch private satellite link for iam url )

**Pull request merged**